purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesAreCompletedBy.kt

```diff
@@ -20,3 +20,12 @@ enum class PurchasesAreCompletedBy {
      */
     MY_APP,
 }
+
+/**
+ * Whether the Purchases SDK should finish transactions.
+ */
+internal val PurchasesAreCompletedBy.finishTransactions: Boolean
+    get() = when (this) {
+        PurchasesAreCompletedBy.REVENUECAT -> true
+        PurchasesAreCompletedBy.MY_APP -> false
+    }
```

Comment:
What are your thoughts on this? Intended as a single source of truth. Related to https://github.com/RevenueCat/purchases-android/pull/1710#issuecomment-2147057244.

Reply:
This is nice! 👍 I like this pattern a lot :) It could also be a parameter for the enum, but IMO this is cleaner.

Reply:
Thanks! Yea I think this is cleaner too. 🙂 Influenced by [extension-oriented design](https://elizarov.medium.com/extension-oriented-design-13f4f27deaee).